scripts/linters/pylint_extensions.py

```diff
@@ -540,11 +651,93 @@ def check_functiondef_params(self, node, node_doc):
             node_doc, node.args, node,
             accept_no_param_doc=node_allow_no_param)
 
-    def check_arg_section_indentation(self, node):
-        """Checks whether the function argument definitions ("Args": section)
-        are indented properly. Parameters should be indented by 4 relative to
-        the 'Arg: ' line and any wrap-around descriptions should be indented
-        by 8. The rest of the docstring section checks are in DocstringChecker.
+    def check_docstring_style(self, node):
+        """It fetches a function node and extract the class node from function
+        node if it is inside a class body and passes it to
+        check_docstring_structure which checks whether the docstring has a
+        space at the beginning and a period at the end.
+
+        Args:
+            node: astroid.scoped_nodes.Function. Node for a function or
+                method definition in the AST.
+        """
+        if node.name in self.constructor_names:
+            class_node = checker_utils.node_frame_class(node)
+            if class_node is not None:
+                self.check_docstring_structure(class_node)
+        self.check_docstring_structure(node)
+
+    def check_newline_above_args(self, node, docstring):
+        """Checks to ensure that there is a single space above the
+        argument parameters in the docstring.
+
+        Args:
+            node: astroid.node.Function. Node for a function or method
+                definition in the AST.
+            docstring: list(str). Function docstring in splitted by newlines.
+        """
+        blank_line_counter = 0
+        for line in docstring:
+            line = line.strip()
+            if line == b'':
+                blank_line_counter += 1
+            if blank_line_counter == 0 or blank_line_counter > 1:
+                if line == b'Args:':
+                    self.add_message(
+                        'single-space-above-args', node=node)
+                elif line == b'Returns:':
+                    self.add_message(
+                        'single-space-above-returns', node=node)
+                elif line == b'Raises:':
+                    self.add_message(
+                        'single-space-above-raises', node=node)
+                elif line == b'Yield:':
+                    self.add_message(
+                        'single-space-above-yield', node=node)
+            if line != b'':
+                blank_line_counter = 0
+
```

Comment:
Aren't we supposed to have a lint check for "methods within a class should be separated by 1 newline, not 2"?

(Again, OK to fix in follow-up PR.)

Reply:
We already have this check listed in #8423

Reply:
OK. You should still fix this instance of it (in your next PR).